src/plugins/data/server/search/strategies/esql_search/esql_search_strategy.ts

```diff
@@ -32,11 +32,12 @@ export const esqlSearchStrategyProvider = (
 
     const search = async () => {
       try {
-        const { terminateAfter, ...requestParams } = request.params ?? {};
+        const { terminateAfter, dropNullColumns, ...requestParams } = request.params ?? {};
+        const queryParams = dropNullColumns ? '?drop_null_columns' : '';
         const { headers, body, meta } = await esClient.asCurrentUser.transport.request(
           {
             method: 'POST',
-            path: '/_query',
+            path: `/_query${queryParams}`,
```

Comment:
nit: why not use the `querystring` param here rather than appending it in the URL?

Reply:
Done https://github.com/elastic/kibana/pull/174585/commits/447e439d77cc0cfa9e307c0301e46c617c7cdd65